### DISCUSSION TITLE
Toast from JavaScript

### DISCUSSION THREAD

**Discussion by ibnIrshad** [livewire/flux#282] [Feature Requests]
When building event handlers for Stripe.js and other third party libraries, I need to call a Flux::toast() from the front end JavaScript, not PHP. Show us how to do that. I'll bet the event handler already exists and it's simply something like `Livewire.dispatch('toast-show', {text})` but I couldn't figure out the exact syntax and it should be documented.

**Comment by ibnIrshad** [comment]
Temporary workaround I have for this is to have a dedicated component next to `<flux:toast />` called `<livewire:toaster />` that handles global events from JavaScript and does toasts.

**Comment by cbautista103** [comment]
I agree with this, and I would also add that there are some other unexpected behaviors of Flux::toast() such as it does not appear to be able to be called from mount().

EDIT: to add on, I can see that there is an event called "toast-show" being dispatched with the following sample parameters: `{duration: 5000, slots: {text: "some text goes here"}}`. But if I manually dispatch that using `Livewire.dispatch`, it does not seem to work.

**Comment by cbautista103** [comment]
@ibnIrshad, and anyone else who may be wondering about this... I was able to come up with a solution. In short, I believe you have to `$dispatch` the toast from a livewire component, not from the global `Livewire` object itself. Here's a solution that allows you to do `Livewire.toast` from any component at any time.

https://github.com/livewire/flux/discussions/505#discussioncomment-11068473

I do feel like there should be a "first-party" way to do this outlined in the docs, but this does the trick for me for now.

**Comment by ibnIrshad** [comment]
I got an answer from Caleb by email, he said that toasts can be shown by dispatching from Alpine: `$dispatch('toast-show', { slots: { text: '?', heading: '?' } })`. Still haven't figured out how to do it "globally" though.

**Reply by cbautista103** [reply]
Take a look at the thread I linked above—that may solve your issue? https://github.com/livewire/flux/discussions/505#discussioncomment-11020634
